autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/AmbariAgentHealthEvaluator.java

```diff
@@ -98,40 +98,39 @@ public void execute() {
             MDCBuilder.buildMdcContext(cluster);
             LOGGER.info("Checking '{}' alerts for cluster {}.", AMBARI_AGENT_HEARTBEAT, cluster.getId());
             AmbariClient ambariClient = ambariClientProvider.createAmbariClient(cluster);
-            List<Map<String, Object>> alertHistory = ambariRequestLogging.logging(() -> ambariClient.getAlert(AMBARI_AGENT_HEARTBEAT_DEF_NAME), "alert");
-            if (!alertHistory.isEmpty()) {
-                List<String> hostNamesToRecover = new ArrayList<>();
-                for (Map<String, Object> history : alertHistory) {
-                    String currentState = (String) history.get(ALERT_STATE);
-                    if (isAlertStateMet(currentState)) {
-                        String hostName = (String) history.get(HOST_NAME);
-                        hostNamesToRecover.add(hostName);
-                        LOGGER.info("Alert: {} is in '{}' state for host '{}'.", AMBARI_AGENT_HEARTBEAT, currentState, hostName);
-                    }
+            List<Map<String, Object>> alerts = ambariRequestLogging.logging(() -> ambariClient.getAlertByNameAndState(
+                    AMBARI_AGENT_HEARTBEAT_DEF_NAME, CRITICAL_ALERT_STATE), "alert[" + AMBARI_AGENT_HEARTBEAT_DEF_NAME + ", " + CRITICAL_ALERT_STATE + "]");
+            List<String> hostNamesToRecover = new ArrayList<>();
+            if (!alerts.isEmpty()) {
```

Comment:
I think this is not necessary, the foreach loop won't do anything on an empty collection.